google/cloud/bigquery/v2/minimal/internal/json_utils.cc

```diff
@@ -24,47 +24,89 @@ void FromJson(std::chrono::milliseconds& field, nlohmann::json const& j,
   auto const l = j.find(name);
   if (l == j.end()) return;
   std::int64_t m;
-  l->get_to(m);
-  field = std::chrono::milliseconds(m);
+  if (l->is_string()) {
+    std::string s;
+    l->get_to(s);
+    char* end;
+    m = std::strtoll(s.c_str(), &end, 10);
+  } else {
+    l->get_to(m);
+  }
+  if (m >= 0) {
+    field = std::chrono::milliseconds(m);
+  }
 }
 
 void ToJson(std::chrono::milliseconds const& field, nlohmann::json& j,
-            char const* name) {
-  j[name] = static_cast<std::int64_t>(
+            char const* name, bool is_number) {
+  auto m = static_cast<std::int64_t>(
```

Comment:
Do you really need to optionally format these as numbers or strings?  Can you always format as strings and assume the receiver will know what to do with them?